sqld/src/admin_api.rs

```diff
@@ -85,23 +90,52 @@ async fn handle_post_block<F: MakeNamespace>(
     }
 }
 
-async fn handle_create_namespace_with_dump<F: MakeNamespace>(
+async fn handle_create_namespace<F: MakeNamespace>(
     State(app_state): State<Arc<AppState<F>>>,
     Path(namespace): Path<String>,
-    body: BodyStream,
+    Json(req): Json<CreateNamespaceReq>,
 ) -> Result<(), crate::error::Error> {
-    let dump = Box::new(body.map_err(|e| std::io::Error::new(ErrorKind::Other, e)));
+    let maybe_dump: Option<DumpStream> =
+        OptionFuture::from(req.dump_url.as_ref().map(dump_stream_from_url))
+            .await
+            .transpose()?;
```

Comment:
Can't this all be done with regular rust? Why do we need OptionFuture? In general I would try to avoid using anything from `futures`.